heroku.md

```diff
@@ -0,0 +1,114 @@
+# What is Heroku
+
+Heroku is a popular all in one hosting solution, you can find more at heroku.com
+
+## Signing Up
+
+You'll need a heroku account, if you don't have one, please sign up here: [https://signup.heroku.com/](https://signup.heroku.com/)
+
+## Installing CLI
+
+Make sure that you've installed the heroku cli tool.
+
+### HomeBrew
+
+```bash
+brew install heroku/brew/heroku
+```
+
+### Other Install Options
+
+See alternative install options here: [https://devcenter.heroku.com/articles/heroku-cli#download-and-install](https://devcenter.heroku.com/articles/heroku-cli#download-and-install).
+
+### Logging in
+
+once you've installed the cli, login with the following:
+
+```bash
+heroku login
+```
+
+### Create an application
+
+Visit dashboard.heroku.com to access your account, and create a new application from the drop down in the upper right hand corner. Heroku will ask a few questions such as region and application name, just follow their prompts.
+
+### Project
+
+Today we're going to be hosting Swift NIO's example http server, you can apply these concepts to your own project. Let's start by cloning NIO
+
+```bash
+
+git clone https://github.com/apple/swift-nio
+```
+
+Make sure to make our newly cloned directory the working directory
+
+
+```bash
+cd swift-nio
+```
+
+By default, Heroku deploys the **master** branch. Always make sure all changes are checked into this branch before pushing.
+
+#### Connect with Heroku
+
+Connect your app with heroku (replace with your app's name).
+
+```bash
+$ heroku git:remote -a your-apps-name-here
+```
+
+### Set Stack
+
+As of 13 September 2018, Heroku’s default stack is Heroku 18, we need it to be 16 for vapor.
+
+```bash
+heroku stack:set heroku-16 -a your-apps-name-here
+```
+
+### Set Buildpack
+
+Set the buildpack to teach heroku how to deal with swift, the vapor-communnity buildpack has all the necessary setup for a swift project.
+
+
+```bash
+heroku buildpacks:set vapor/vapor
```

Comment:
Maybe we should clarify here that this is Vapor's _Swift_ buildpack? Otherwise I could see people assuming this somehow installs or depends on the Vapor framework.

Reply:
+1